astropy/cosmology/__init__.py

```diff
@@ -8,27 +8,40 @@
 
 """
 
-from . import core, flrw, funcs, parameter, units, utils
+from . import flrw, units
 
 from . import io  # needed before 'realizations'  # isort: split
 from . import realizations
-from .core import *
+from .core import Cosmology, CosmologyError, FlatCosmologyMixin
 from .flrw import *
-from .funcs import *
-from .parameter import *
+from .funcs import cosmology_equal, z_at_value
+from .parameter import Parameter
 from .realizations import available, default_cosmology
-from .utils import *
 
-__all__ = (
-    core.__all__
-    + flrw.__all__  # cosmology classes
-    + realizations.__all__  # instances thereof
-    + ["units"]
-    # utils
-    + funcs.__all__
-    + parameter.__all__
-    + utils.__all__
-)
+__all__ = [
+    "Cosmology",
+    "FlatCosmologyMixin",
+    "Parameter",
+    "CosmologyError",
+    # realizations
+    "available",
+    "default_cosmology",
+    "WMAP1",  # In `__getattr__`
+    "WMAP3",  # In `__getattr__`
+    "WMAP5",  # In `__getattr__`
+    "WMAP7",  # In `__getattr__`
+    "WMAP9",  # In `__getattr__`
+    "Planck13",  # In `__getattr__`
+    "Planck15",  # In `__getattr__`
+    "Planck18",  # In `__getattr__`
+    # funcs
+    "z_at_value",
+    "cosmology_equal",
+    # public modules
+    "realizations",
+    "units",
```

Comment:
Could this be built out of the `__public_api__` in the other modules?

Reply:
Yes, except ``__all__`` is special-cased in mypy and static type-checkers and I don't think dynamically building it from ``__public_api__`` will be compatible.

Reply:
I've remove `__public_api__` from this PR.